hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaTargetManagement.java

```diff
@@ -371,27 +343,44 @@ public void deleteByControllerID(final String controllerID) {
     }
 
     @Override
-    public Page<Target> findByAssignedDistributionSet(final Pageable pageReq, final long distributionSetID) {
-        throwEntityNotFoundIfDsDoesNotExist(distributionSetID);
+    public Page<Target> findByAssignedDistributionSet(final Pageable pageReq, final long distributionSetId) {
+        throwEntityNotFoundIfDsDoesNotExist(distributionSetId);
+        return targetRepository.findByAssignedDistributionSetId(pageReq, distributionSetId);
+    }
 
-        return targetRepository.findByAssignedDistributionSetId(pageReq, distributionSetID);
+    @Override
+    public Page<Target> findByAssignedDistributionSetAndQuery(final Pageable pageable, final long distributionSetId,
+            final String query) {
+        throwEntityNotFoundIfDsDoesNotExist(distributionSetId);
+        final Set<String> controllerIds = targetRepository.findControllerIdsByAssignedDistributionSet(distributionSetId);
```

Comment:
yes - pageable doesn't help - I need the complete list of IDs

Reply:
`targetRepository.findControllerIdsByAssignedDistributionSet(distributionSetId)` may produce a pretty big result list. Better split this into chunks (let's say 500 IDs per call) and page through the list to fetch all. Otherwise this will put a lot of load on the DB

Reply:
- I do not throw away the ID-list, I pass it to the query-service in order to get a proper-paged result back
- I do respect the pageable request by passing it to the query-service
- I have no idea what I should do with only a subset of IDs in the "AND-condition"

Please suggest an implementation which can do better.